### ISSUE TITLE
Narrow down what the controller SA can 'update'

### ISSUE THREAD

**Issue by sbose78** [redhat-developer/service-binding-operator#454]
I would like to propose that we must reduce the following [permission](https://github.com/redhat-developer/service-binding-operator-manifests/blob/master/0.1.1-293/service-binding-operator.v0.1.1-293.clusterserviceversion.yaml#L145) to allow `update` of only those resources which we know for sure are going to have the binding secret injected. 

```
- apiGroups:
          - '*'
          resources:
          - '*'
          verbs:
          - get
          - list
          - watch
          - update
```

That way we'll be able to reduce the privileges of the service account to have permissions to `update` popular podSpec-based workloads and any other well-known resource types.

- deployments
- deploymentconfigs
- daemonsets
- replicasets
- statefulsets
- services ( knative )

We only need to be able to get, read and watch all resources because we are dealing with CRDs and we don't know in advance what is to be used for binding.

**Comment by sbose78** [comment]
cc @redhat-developer/service-binding

**Comment by sbose78** [comment]
Note: Changes should be made in this repo's /deploy directory.

**Comment by pedjak** [comment]
@sbose78 how about that we drop the above block from `role.yaml` and then define an additional rolebinding for the service account:

```yaml
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: service-binding-operator
subjects:
- kind: ServiceAccount
  name: service-binding-operator
roleRef:
  kind: ClusterRole
  name: edit
  apiGroup: rbac.authorization.k8s.io
```

`edit` is aggregating cluster role, so we automatically get access to other resources as well, if the other operators behave properly.

**Comment by sbose78** [comment]
@pedjak +1

**Comment by pedjak** [comment]
Fixed by #979 and #1002
